### PR TITLE
Updated to cover box and sign

### PR DESCRIPTION
**Pull request by cbowdon**
I've added a few more Kalium wrappers:

- box
- sign
- sha256 & sha512
- hexify

I've also updated to the latest stable version of Kalium and Clojure. Let me know if this is ok.

### CODE REVIEW

**Comment by cbowdon** [comment]
Hmm, looks like the CI build failed because apt-get upgrade couldn't upgrade Postgres while it's still running. Adding a build step to stop Postgres would work, but then it could be whack-a-mole with every other package that won't upgrade smoothly. Would not upgrading the system work? (only upgrading relevant packages for the libsodium deb)

**Comment by lvh** [comment]
Somehow, this didn't notify me :) I'm looking over the PR now.

File: test/caesium/crypto/generichash_test.clj [hunk 1]
```diff
@@ -33,3 +33,21 @@
...
+
+(def empty-string (.getBytes ""))
+
+(def sha256-message (.getBytes "My Bonnie lies over the ocean, my Bonnie lies over the sea"))
+(def sha256-digest (unhexify "d281d10296b7bde20df3f3f4a6d1bdb513f4aa4ccb0048c7b2f7f5786b4bcb77"))
+(def sha256-empty-string-digest (unhexify "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855"))
+
+(def sha512-message (.getBytes "My Bonnie lies over the ocean, Oh bring back my Bonnie to me"))
+(def sha512-digest (unhexify "2823e0373001b5f3aa6db57d07bc588324917fc221dd27975613942d7f2e19bf444654c8b9f4f9cb908ef15f2304522e60e9ced3fdec66e34bc2afb52be6ad1c"))
+(def sha512-empty-string-digest (unhexify "cf83e1357eefb8bdf1542850d66d8007d620e4050b5715dc83f4a921d36ce9ce47d0d13c5d85f2b0ff8318d2877eec2f63b931bd47417a81a538327af927da3e"))
```
Comment (lvh) [inline test/caesium/crypto/generichash_test.clj]: Did you generate these from a known-good source, or are these a test vector?

Comment (cbowdon) [inline test/caesium/crypto/generichash_test.clj]: These are test vectors taken from https://github.com/abstractj/kalium/commit/e0e87455e8f791a1553bdaabf21c3acb3ca310d6 .

Is that a bad idea? I guess it means the test shows it is wrapping kalium correctly, but not that the functionality is actually correct.

**Comment by lvh** [comment]
I'll make the necessary changes in master, but you'll need to merge master back in to incorporate them.

**Comment by lvh** [comment]
As a side note (and it's OK to not have an opinion here); do you think that in the future it might be better to just call appropriate methods directly through jffi, instead of going through the "convenience" objects Kalium provides? It seems like everything is just about subverting that OO layer.

**Comment by cbowdon** [comment]
Thanks for taking the time to review this. I'll rebase my changes on master.

I have no experience of jffi, but think it would be better to lose the kalium dependency if possible. Besides subverting the OO layer there is the issue that kalium doesn't cover everything libsodium provides.

**Comment by cbowdon** [comment]
Sorry I should have commented - I have rebased on top of the changes so this branch now passes the CI checks.

**Comment by lvh** [comment]
This looks great; thanks!

**Pull request merged**